docs/faq.rst

```diff
@@ -493,6 +493,37 @@ maybe open an issue in their issue tracker. Do not file an issue in the
 If you can reproduce the issue with the proven filesystem, please file an
 issue in the |project_name| issue tracker about that.
 
+Why does running 'borgbackup check --repair' warn about data loss?
+------------------------------------------------------------------
+
+Due to the nature of backup, and especially deduplicated backups as used
+by |project_name|, some repair operations may require a roll-back to an
+earlier version of the repository. Some errors, especially where data
```

Comment:
I will review this further. I tried to understand your response to the original ticket. My first thought was that the repair function was relatively new, based on what you said, and that was the cause for the warning. 

However, after reviewing the code and several updates where Repair was implemented, that didn't make sense.  So when I wrote this, I approached the warning as making a statement that --repair will change the repository. Generally, it changes the repository for the better. Sometimes, in some unforeseen instances, it may make the repository worse.  Hence the warning and why I try to make clear that repair should be performed on a repo copy.

From what I have been able to glean from the code, repair will:
- Repair broken indexes (this requires rewriting the index, correct?).
- Attempt to correct corrupted segment files if checksums exist (which would imply modifying such segment files)